### PR TITLE
Add build context parameter to docker build

### PR DESCRIPTION
**Pull request by patricoferris**
Currently the `Current_docker.Default.build` function allows a user to specify the relative path to a Dockerfile from the Git src, but there's no way as far as I can tell to specify the build context path. So if I have a Dockerfile in `src` that assumes the build context is `src` I can't run a build in that context.

The current implementation does potentially allow a user to escape the build directory though. I don't know if this is too much of a problem if the user is always specifying this path and maybe it is up to the user to check the `~path` variable is not trying to do something bad?

### CODE REVIEW

**Review by maiste — APPROVED** [review]
LGTM :+1: 
@patricoferris, is it ok if I force-push your PR (after a rebase on master) before merging?
(If you allow edit by maintainers)

**Comment by patricoferris** [comment]
Thanks for the review :)) Fine by me !

**Comment by maiste** [comment]
I rebased on master. However, I prefer to have a second point of view for the escaping `path` issue before merging something that doesn't fit with the `ocurrent` team's point of view. As you question, do we want the user to have the possibility to escape from the build directory and make the user responsible for that or not?

**Comment by talex5** [comment]
> As you question, do we want the user to have the possibility to escape from the build directory and make the user responsible for that or not?

Who is the "user" here? The author of the pipeline, or the person submitting the PR?

The case we want to prevent is someone making a PR in which `src` is a symlink to `/root` or something.

**Comment by maiste** [comment]
In this case, can we consider this is something that `ocurrent` doesn't have to take care of as it will be used in the pipeline code?
If yes, maybe it is worth adding a warning about it in the documentation.

**Comment by maiste** [comment]
I rebase over master and add a small warning. Do you think we can merge it?

File: plugins/docker/s.ml
Comment (tmcgilchrist) [inline plugins/docker/s.ml:32]: Should this be a comment?

Comment (maiste) [reply plugins/docker/s.ml]: It shouldn't even be here. Thanks.

**Pull request merged**
